### PR TITLE
Make JDBC install guide more user-friendly

### PR DESCRIPTION
**Pull request by m57lyra**
This replaces PR 4992.

### CODE REVIEW

**Comment by m57lyra** [comment]
I'm fairly sure I know what happened - there was a change to the file for standardizing on the "class path" spelling between when I started this work and today.

**Comment by m57lyra** [comment]
This PR replaces https://github.com/prestosql/presto/pull/4992. See that PR for the comments that were resolved if needed.

**Review by mosabua — APPROVED** [review]

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 1]
```diff
@@ -2,7 +2,20 @@
 JDBC Driver
 ===========
 
-Presto can be accessed from Java using the JDBC driver.
+The Presto JDBC (Java Database Connectivity) driver allows users to access
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:5]: I would assume people commonly use "JDBC driver" without expanding the acronym, so expanding it here can make the text less clear, not more.
What about we make "JDBC driver" a link to a webpage explaning what JDBC is?

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 2]
```diff
@@ -2,7 +2,20 @@
 JDBC Driver
 ===========
 
-Presto can be accessed from Java using the JDBC driver.
+The Presto JDBC (Java Database Connectivity) driver allows users to access 
+Presto using Java-based applications. Both desktop and server-side
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:6]: > Java-based

And other non-Java applications running on JVM.

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 3]
```diff
@@ -2,7 +2,20 @@
...
-Presto can be accessed from Java using the JDBC driver.
+The Presto JDBC (Java Database Connectivity) driver allows users to access 
+Presto using Java-based applications. Both desktop and server-side 
+applications, such as those used for reporting and database development, use 
+the JDBC driver.
+
+Requirements
+------------
+
+The JDBC driver is compatible with Java 8+, and can be used with applications
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:13]: > 8+

this looks informal. maybe "8 and later"?

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 4]
```diff
@@ -15,44 +28,63 @@ The driver is also available from Maven Central:
         <version>\ |version|\ </version>
     </dependency>
 
-Requirements
-------------
+You should be using the same version of the JDBC driver as your Presto
+installation. If you are not on the latest release shown above, a list of all
+currently available versions can be found `here <https://repo1.maven.org/maven2/io/prestosql/presto-jdbc>`__.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:33]: This page is not the nicest one. Maybe we link to https://search.maven.org/artifact/io.prestosql/presto-jdbc instead?

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 5]
```diff
@@ -15,44 +28,63 @@ The driver is also available from Maven Central:
...
 
-The JDBC driver is compatible with Java 8, and can be used with applications
-running on Java virtual machines version 8 and higher.
+Once downloaded, you must add the JAR file to the classpath of your Java 
+application. Often, this will be the lib folder of a Java application, but
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:36]: I'd not mention "the lib folder".

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 6]
```diff
@@ -15,44 +28,63 @@ The driver is also available from Maven Central:
...
+you should check the documentation for your specific application. Depending 
+on your application, you may be able to accomplish this from a dialog rather 
+than on the command line.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:39]: Since you do not provide any exact or template command line instructions, I would remove the sentence about dialog/cli

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 7]
```diff
@@ -15,44 +28,63 @@ The driver is also available from Maven Central:
...
+------------------------------------
 
-Connecting
-----------
+Drivers are commonly automatically loaded by applications once they are added 
+to its classpath. If your application does not, this section will be helpful.
 
-The following JDBC URL formats are supported:
+If your application has a UI, look for a Settings option or a Drivers tab and 
+register the JDBC driver there. The steps to register the JDBC
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:53]: How many (%) applications have "Drivers" tab in "Settings"?

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 8]
```diff
@@ -15,44 +28,63 @@ The driver is also available from Maven Central:
...
+connect to Presto from your application. The following example illustrates
+using a JDBC URL to create a connection:
 
 .. code-block:: java
 
     String url = "jdbc:presto://example.net:8080/hive/sales";
     Connection connection = DriverManager.getConnection(url, "test", null);
 
+This example JDBC URL locates a Presto instance running on port ``8080`` in
+``example.net``, with the catalog ``hive`` and the schema ``sales`` defined.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:78]: "in example.net" or "on example.net"?

File: presto-docs/src/main/sphinx/installation/jdbc.rst (hunk 1 above)
Comment (m57lyra) [reply presto-docs/src/main/sphinx/installation/jdbc.rst:5]: Great idea. Almost every decent page is either an obscure blog (ie, not really reference material), or talks about someone else's driver in particular. Are we ok with wikipedia links in general? https://en.wikipedia.org/wiki/JDBC_driver

Comment (findepi) [reply presto-docs/src/main/sphinx/installation/jdbc.rst:5]: I am fine. https://docs.oracle.com/javase/tutorial/jdbc/overview/index.html may be useful too

Comment (mosabua) [reply presto-docs/src/main/sphinx/installation/jdbc.rst:5]: Both are okay.

File: presto-docs/src/main/sphinx/installation/jdbc.rst (hunk 7 above)
Comment (m57lyra) [reply presto-docs/src/main/sphinx/installation/jdbc.rst:53]: Using the definitive guide as a model, touching on this briefly here.

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 9]
```diff
@@ -2,7 +2,20 @@
 JDBC Driver
 ===========
 
-Presto can be accessed from Java using the JDBC driver.
+The Presto `JDBC driver <https://en.wikipedia.org/wiki/JDBC_driver>`__ allows users to access Presto using
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:5]: one underline for links

line wrapping

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 10]
```diff
@@ -15,44 +28,60 @@ The driver is also available from Maven Central:
         <version>\ |version|\ </version>
     </dependency>
 
-Requirements
-------------
+You should be using the same version of the JDBC driver as your Presto
+installation. If you are not on the latest release shown above, a list of all
+currently available versions can be found `here <https://search.maven.org/artifact/io.prestosql/presto-jdbc>`__.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:33]: Dont use "here". Instead use "the Maven Central repository" 

And use this link in fact - https://repo1.maven.org/maven2/io/prestosql/presto-jdbc/ .. it allows browsing

Also mention that the folder contains all the different version and in each folder the driver file is named presto-jdbc-xxx.jar  .. with xxx the version number .. that is the file that needs to be downloaded .. not any of the others.

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 11]
```diff
@@ -15,44 +28,60 @@ The driver is also available from Maven Central:
...
 
-The JDBC driver is compatible with Java 8, and can be used with applications
-running on Java virtual machines version 8 and higher.
+Once downloaded, you must add the ``.jar`` file to a directory in the classpath
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:35]: you can say .. the JAR file

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 12]
```diff
@@ -15,44 +28,60 @@ The driver is also available from Maven Central:
...
 
-The JDBC driver is compatible with Java 8, and can be used with applications
-running on Java virtual machines version 8 and higher.
+Once downloaded, you must add the ``.jar`` file to a directory in the classpath 
+of users on systems where they will access Presto. Alternatively, you can copy
+the ``.jar`` file to its own directory and add that directory to the classpath.
 
-Driver Name
------------
+Once you have downloaded the JDBC driver and added it to your
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:39]: Maybe rephrase to avoid duplicate "Once.."

File: presto-docs/src/main/sphinx/installation/jdbc.rst [hunk 13]
```diff
@@ -15,44 +28,60 @@ The driver is also available from Maven Central:
...
 
-The driver class name is ``io.prestosql.jdbc.PrestoDriver``.
-Most users do not need this information as drivers are loaded automatically.
+Registering & configuring the driver
+------------------------------------
 
-Connecting
-----------
+Drivers are commonly automatically loaded by applications once they are added 
+to its classpath. If your application does not, this section will be helpful.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/installation/jdbc.rst:48]: reword to avoid future tense

**Review by mosabua — CHANGES_REQUESTED** [review]
A bunch of clarifications needed.

**Review by mosabua — APPROVED** [review]
Nice work.

**Review by electrum — APPROVED** [review]
